scene/debugger/scene_debugger.cpp

```diff
@@ -1075,4 +1175,902 @@ void LiveEditor::_reparent_node_func(const NodePath &p_at, const NodePath &p_new
+				return;
+			}
+		} else if (node_select_type == NODE_TYPE_3D) {
+#ifndef _3D_DISABLED
+			if (root->get_camera_3d() && _handle_3d_input(p_event)) {
+				return;
+			}
+#endif // _3D_DISABLED
+		}
+	}
+
+	if (node_select_type == NODE_TYPE_NONE || selection_list->is_visible()) {
+		return;
+	}
+
+	Ref<InputEventMouseButton> b = p_event;
+	if (!b.is_valid() || !b->is_pressed()) {
+		return;
+	}
+
+	bool list_shortcut = node_select_mode == SELECT_MODE_SINGLE && b->get_button_index() == MouseButton::RIGHT && b->is_alt_pressed();
+	if (!list_shortcut && b->get_button_index() != MouseButton::LEFT) {
+		return;
+	}
+
+	Vector<SelectResult> items;
+	if (node_select_type == NODE_TYPE_2D) {
+		Transform2D canvas_override = camera_override ? root->get_canvas_transform_override() : root->get_canvas_transform();
+		Point2 pos = (canvas_override * root->get_screen_transform()).affine_inverse().xform(b->get_position());
+		for (int i = 0; i < root->get_child_count(); i++) {
+			_find_canvas_items_at_pos(pos, root->get_child(i), items);
+		}
+
+		// Remove possible duplicates.
+		for (int i = 0; i < items.size(); i++) {
+			Node *item = items[i].item;
+			for (int j = 0; j < i; j++) {
+				if (items[j].item == item) {
+					items.remove_at(i);
+					i--;
+
+					break;
+				}
+			}
+		}
+	} else if (node_select_type == NODE_TYPE_3D) {
+#ifndef _3D_DISABLED
+		Point2 pos = (root->get_screen_transform()).affine_inverse().xform(b->get_position());
+		_find_3d_items_at_pos(pos, items);
+#endif // _3D_DISABLED
+	}
+
+	if (items.is_empty()) {
+		return;
+	}
+
+	items.sort();
+
+	if ((!list_shortcut && node_select_mode == SELECT_MODE_SINGLE) || items.size() == 1) {
+		Array message;
+		message.append(items[0].item->get_instance_id());
+		EngineDebugger::get_singleton()->send_message("remote_node_clicked", message);
+	} else if (list_shortcut || node_select_mode == SELECT_MODE_LIST) {
+		if (!selection_list->is_inside_tree()) {
+			root->add_child(selection_list);
+		}
+
+		selection_list->clear();
+		for (SelectResult &I : items) {
+			selection_list->add_item(I.item->get_name());
+			selection_list->set_item_metadata(-1, I.item);
+		}
+
+		selection_list->set_position(selection_list->is_embedded() ? b->get_position() : b->get_position() + root->get_position());
+		selection_list->reset_size();
+		selection_list->popup();
+		// FIXME: Ugly hack that stops the popup from hiding when the button is released.
+		selection_list->call_deferred(SNAME("set_position"), selection_list->get_position() + Point2(1, 0));
+	}
+}
+
+void RuntimeNodeSelect::_items_popup_index_pressed(int p_index, PopupMenu *p_popup) {
+	Array message;
+	message.append(Object::cast_to<Node>(p_popup->get_item_metadata(p_index))->get_instance_id());
+	EngineDebugger::get_singleton()->send_message("remote_node_clicked", message);
+}
+
+void RuntimeNodeSelect::_update_input_state() {
+	SceneTree *scene_tree = SceneTree::get_singleton();
+	if (!scene_tree->get_root()->is_inside_tree()) {
```

Comment:
??
How the root can be outside tree?

Reply:
At the very beginning. Without this check the game process crashes on launch.

Reply:
That's weird, but ok. Maybe leave a comment.